tools/inference_engine/src/impl/node.h

```diff
@@ -73,6 +79,14 @@ namespace inference_engine
             outputs_.push_back(node);
         }
 
+        virtual void replace_input(const GpuNode* old_node, GpuNode* new_node)
```

Comment:
can we refactor it to remove_input(..node)
a than have some utility, helper function outside of this class which is doing replace_input logic?